### PR TITLE
ELEMENTS-1283: fix nuxeo page provider params management

### PR DESCRIPTION
**Pull request by vpasquier**
Same than https://github.com/nuxeo/nuxeo-elements/pull/318

### CODE REVIEW

**Review by Gabez0r — APPROVED** [review]

**Review by mnixo — APPROVED** [review]

File: core/nuxeo-page-provider.js [hunk 1]
```diff
@@ -428,26 +428,26 @@ import './nuxeo-resource.js';
...
+          if (queryParams) {
+            params.queryParams = queryParams;
           }
-        }, this);
-        params.namedParameters = namedParams;
-      }
+          return;
+        }
+        let value = this.params[key];
+        if (value != null) {
```
Comment (andreacornaglia) [inline core/nuxeo-page-provider.js:441]: i suggest either `if(value)` or `if (value !== null)`

File: core/nuxeo-page-provider.js [hunk 2]
```diff
@@ -428,26 +428,26 @@ import './nuxeo-resource.js';
...
+          if (Array.isArray(value)) {
+            value = JSON.stringify(value.map((item) => (item['entity-type'] ? item.uid || item.id : item)));
```
Comment (andreacornaglia) [inline core/nuxeo-page-provider.js:443]: i suggest `item['entity-type'] && (item.uid || item.id) || item`

File: core/nuxeo-page-provider.js (hunk 1 above)
Comment (andreacornaglia) [reply core/nuxeo-page-provider.js:441]: also to avoid complexity, maybe if(!value) return

Comment (Gabez0r) [reply core/nuxeo-page-provider.js:441]: Please note that what you suggested is no equivalent to what we have here. `value != null` is used to check for both not `undefined` or not `null`, whereas just checking for `value` would also also check for no empty strings.

**Pull request merged**